renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -0,0 +1,352 @@
+                              (void **)&header);
+  m_pDriver->CheckVkResult(vkr);
+
+  // invalidate the cpu cache for this memory range to avoid reading stale data
+  const VkMappedMemoryRange range = {
+      VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE,
+      NULL,
+      Unwrap(result.alloc.mem),
+      result.alloc.offs,
+      size,
+  };
+  vkr = ObjDisp(d)->InvalidateMappedMemoryRanges(Unwrap(d), 1, &range);
+  m_pDriver->CheckVkResult(vkr);
+
+  // Count the BLAS device addresses to update the AS type
+  const uint64_t handleCount = *(uint64_t *)(header + handleCountOffset);
+  result.isTLAS = handleCount > 0;
+
+  ObjDisp(d)->UnmapMemory(Unwrap(d), Unwrap(result.alloc.mem));
+
+  return true;
+}
+
+template <typename SerialiserType>
+bool VulkanAccelerationStructureManager::Serialise(SerialiserType &ser, ResourceId id,
+                                                   const VkInitialContents *initial,
+                                                   CaptureState state)
+{
+  VkDevice d = !IsStructuredExporting(state) ? m_pDriver->GetDev() : VK_NULL_HANDLE;
+  const bool replayingAndReading = ser.IsReading() && IsReplayMode(state);
+  VkResult vkr = VK_SUCCESS;
+
+  byte *contents = NULL;
+  uint64_t contentsSize = initial ? initial->mem.size : 0;
+  MemoryAllocation mappedMem;
+
+  // Serialise this separately so that it can be used on reading to prepare the upload memory
+  SERIALISE_ELEMENT(contentsSize);
+
+  const VkDeviceSize nonCoherentAtomSize = m_pDriver->GetDeviceProps().limits.nonCoherentAtomSize;
+
+  // the memory/buffer that we allocated on read, to upload the initial contents.
+  MemoryAllocation uploadMemory;
+  VkBuffer uploadBuf = VK_NULL_HANDLE;
+
+  if(ser.IsWriting())
+  {
+    if(initial && initial->mem.mem != VK_NULL_HANDLE)
+    {
+      const VkDeviceSize size = AlignUp(initial->mem.size, nonCoherentAtomSize);
+
+      mappedMem = initial->mem;
+      vkr = ObjDisp(d)->MapMemory(Unwrap(d), Unwrap(mappedMem.mem), initial->mem.offs, size, 0,
+                                  (void **)&contents);
+      m_pDriver->CheckVkResult(vkr);
+
+      // invalidate the cpu cache for this memory range to avoid reading stale data
+      const VkMappedMemoryRange range = {
+          VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE, NULL, Unwrap(mappedMem.mem), mappedMem.offs, size,
+      };
+
+      vkr = ObjDisp(d)->InvalidateMappedMemoryRanges(Unwrap(d), 1, &range);
+      m_pDriver->CheckVkResult(vkr);
+    }
+  }
+  else if(IsReplayMode(state) && !ser.IsErrored())
+  {
+    // create a buffer with memory attached, which we will fill with the initial contents
+    const VkBufferCreateInfo bufInfo = {VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO, NULL, 0, contentsSize,
+                                        VK_BUFFER_USAGE_TRANSFER_SRC_BIT |
+                                            VK_BUFFER_USAGE_TRANSFER_DST_BIT |
+                                            VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT};
```

Comment:
Minor nit: if you add a trailing `,` to larger struct definitions like this then clang-format will do a better job. No trailing `,` is usually most useful for struct/array definitions that are small enough to fit in one line or so.